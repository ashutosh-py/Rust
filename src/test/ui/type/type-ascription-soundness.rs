// Type ascription doesn't lead to unsoundness

#![feature(type_ascription)]

fn main() {
    let arr = &[1u8, 2, 3];
    let ref x = arr: &[u8];
      //~^ ERROR type ascriptions are not
    let ref mut x = arr: &[u8];
      //~^ ERROR type ascriptions are not
    match arr: &[u8] {
      //~^ ERROR type ascriptions are not
        ref x => {}
    }
    let _len = (arr: &[u8]).len();
      //~^ ERROR type ascriptions are not
}
