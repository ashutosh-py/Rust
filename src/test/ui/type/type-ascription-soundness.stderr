error: type ascriptions are not allowed in lvalue contexts
  --> $DIR/type-ascription-soundness.rs:7:17
   |
LL |     let ref x = arr: &[u8];
   |                 ^^^^^^^^^^
   |
help: try to use the type ascription when creating the local variable
   |
LL | let arr: &[u8] = &[1u8, 2, 3];
   | ^^^^^^^^^^^^^^

error: type ascriptions are not allowed in lvalue contexts
  --> $DIR/type-ascription-soundness.rs:8:21
   |
LL |     let ref mut x = arr: &[u8];
   |                     ^^^^^^^^^^
   |
help: try to use the type ascription when creating the local variable
   |
LL | let arr: &[u8] = &[1u8, 2, 3];
   | ^^^^^^^^^^^^^^

error: type ascriptions are not allowed in lvalue contexts
  --> $DIR/type-ascription-soundness.rs:9:11
   |
LL |     match arr: &[u8] {
   |           ^^^^^^^^^^
   |
help: try to use the type ascription when creating the local variable
   |
LL | let arr: &[u8] = &[1u8, 2, 3];
   | ^^^^^^^^^^^^^^

error: type ascriptions are not allowed in lvalue contexts
  --> $DIR/type-ascription-soundness.rs:9:11
   |
LL |     match arr: &[u8] {
   |           ^^^^^^^^^^
   |
help: try to use the type ascription when creating the local variable
   |
LL | let arr: &[u8] = &[1u8, 2, 3];
   | ^^^^^^^^^^^^^^

error: type ascriptions are not allowed in lvalue contexts
  --> $DIR/type-ascription-soundness.rs:12:16
   |
LL |     let _len = (arr: &[u8]).len();
   |                ^^^^^^^^^^^^
   |
help: try to use the type ascription when creating the local variable
   |
LL | let arr: &[u8] = &[1u8, 2, 3];
   | ^^^^^^^^^^^^^^

error: aborting due to 5 previous errors

